clippy_lints/src/types.rs

```diff
@@ -267,6 +295,27 @@ impl Types {
                 let res = qpath_res(cx, qpath, hir_id);
                 if let Some(def_id) = res.opt_def_id() {
                     if Some(def_id) == cx.tcx.lang_items().owned_box() {
+                        if_chain! {
+                            let last = last_path_segment(qpath);
+                            if let Some(ref params) = last.args;
+                            if !params.parenthesized;
+                            if let Some(ty) = params.args.iter().find_map(|arg| match arg {
+                                GenericArg::Type(ty) => Some(ty),
+                                _ => None,
+                            });
+                            if let TyKind::Rptr(ref _lifetime_opt, ref _mut) = ty.kind;
```

Comment:
```suggestion
                            if let TyKind::Rptr(..) = ty.kind;
```

Rather use the rest pattern if you don't use the fields at all.

Reply:
sure, there could be a lint for this too.